activation/activation.go

```diff
@@ -65,35 +93,29 @@ func (b Builder) BuildActivationTx(nipst *nipst.NIPST) error {
 
 func (b *Builder) GetPrevAtxId(node mesh.NodeId) (*mesh.AtxId, error) {
 	ids, err := b.db.GetNodeAtxIds(node)
-	if err != nil {
-		return nil, err
-	}
-	if len(ids) == 0 {
-		return nil, nil
+
+	if len(ids) == 0 || err != nil {
+		return &mesh.EmptyAtx, err
 	}
 	return &ids[len(ids)-1], nil
 }
 
-func (b *Builder) GetPositioningAtx(l mesh.LayerID) (*mesh.AtxId, error) {
-	atxs, err := b.db.GetLayerAtxIds(l)
+func (b *Builder) GetPositioningAtxId(epochId EpochId) (*mesh.AtxId, error) {
+	atxs, err := b.db.GetEpochAtxIds(epochId)
 	if err != nil {
 		return nil, err
 	}
-	if len(atxs) == 0 {
-		//is this so?
-		return nil, nil
-	}
 	atxId := atxs[rand.Int31n(int32(len(atxs)))]
 
 	return &atxId, nil
 }
 
 func (b *Builder) GetLastSequence(node mesh.NodeId) uint64 {
 	atxId, err := b.GetPrevAtxId(node)
-	if err != nil {
+	if *atxId == mesh.EmptyAtx {
```

Comment:
why not compare to null pointer ?

Reply:
I'd also check the error beforehand and assume atxId is nil/empty if err is not nil.

Reply:
also does error always imply the seq number is 0 ? or maybe there's a real error that we need to take care of ?